fortuna/data/dataset/huggingface_datasets.py

```diff
@@ -0,0 +1,207 @@
+import abc
+import logging
+
+from typing import Union, Dict, List, Iterable, Optional, Tuple, Sequence, Any
+from tqdm import tqdm
+import jax.random
+from datasets import DatasetDict, Dataset
+from jax import numpy as jnp
+from jax.random import PRNGKeyArray
+from transformers import PreTrainedTokenizer, BatchEncoding, DataCollatorWithPadding, DataCollatorForLanguageModeling
+from transformers.utils import PaddingStrategy
+
+from fortuna.data.loader.huggingface_loaders import HuggingFaceDataLoader
+from fortuna.data.loader.utils import IterableData
+from fortuna.typing import Array
+
+logger = logging.getLogger(__name__)
+
+
+class HuggingFaceClassificationDatasetABC(abc.ABC):
+    def __init__(
+        self,
+        tokenizer: PreTrainedTokenizer,
+        max_length: int,
+        padding: Union[bool, str, PaddingStrategy] = True,
+        per_device_train_batch_size: int = 8,
+        per_device_eval_batch_size: int = 8,
```

Comment:
Is this `train` and `eval` distinction still relevant now that you don't offer any longer explicit training and test loaders?

Reply:
Would it be better to offer just one, and the user calls the object for each data loader (train, val, calib, test)?

Reply:
Yes the distinction is relevant because in `eval` you can use larger bach size given that there is no backward pass.